pkg/printers/humanreadable.go

```diff
@@ -631,6 +538,88 @@ func (h *HumanReadablePrinter) PrintTable(obj runtime.Object, options PrintOptio
 	return table, nil
 }
 
+// printRowsForHandlerEntry prints the incremental table output (headers if the current type is
+// different from lastType) including all the rows in the object. It returns the current type
+// or an error, if any.
+func printRowsForHandlerEntry(output io.Writer, handler *handlerEntry, obj runtime.Object, options PrintOptions, lastType reflect.Type) (reflect.Type, error) {
+	t := reflect.TypeOf(obj)
+	if !options.NoHeaders && t != lastType {
```

Comment:
So I think the right solution here is not to pass individual unstructured down into print obj, but to pass tables down.  And Get should be trying to convert responses from the server into tabular output.  We don't want two different unstructured from a single list to have different columns, and PrintObj can now accept tables.

So as the replacement for the previous dynamic logic, in 1.8 Get will take on the responsibility for turning server lists into Tables and passing them to PrintObj.  That means that the default handler for unstructured won't be used anymore.